_posts/2017-12-06-google-assistant.md

```diff
@@ -0,0 +1,50 @@
+---
+layout: post
+title: 'Apps ontwikkelen voor de Google Assistant'
+category: Google Assistant
+tags: Google, Assistant, Actions on Google, Dialogflow, Firebase,
+comments: true
+author: bart
+min_read: 5
+---
+De Google Assistant, de digitale voice assistant van Google. In de US is het al een groot platform, en nu met de toevoeging van nieuwe ondersteunde talen begint het ook op te komen in andere landen. De meeste Move4Mobile medewerkers hebben ondertussen wel eens geprobeerd een gesprek aan te gaan met de Google Home, of werden juist gek van een ander die weer tegen de speaker stond te praten. Vaak gaat het gesprek niet verder dan een paar simpele voor gedefineerde commando’s, maar zo'n Assistant kan veel meer dan dat!
+
+### Het Assistant platform
+
+De Google Assistant is één van de vele voice assistants, van al deze assistants zijn Apple's Siri, Amazon's Alexa en Google's Assistant de meest populaire. Siri is echter niet uitbreidbaar met persoonlijke apps en Alexa is tot op heden alleen beschikbaar in de US. 
+De Assistant staat open voor nieuwe apps en ondersteund steeds meer talen, ook biedt deze vele functies aan, waaronder het bijhouden van een shopping-list, Netflix streamen naar je tv en simpele acties zoals het zetten van een alarm. Over deze functies ga ik het echter niet hebben, waar ik mij op ga richten is wat de Assistant nog niet uit zichzelf kan doen, maar wa jij de Assistant kan laten doen. Wanneer je wilt verbinden met een bestaande service of een apparaat dat nog niet ondersteunt wordt is het mogelijk om zelf een app te schrijven voor de Assistant. Google biedt de mogelijkheid om je eigen "chat bot" app toe te voegen aan de Assistant. Wanneer je app eenmaal in de store staat kun je deze aanroepen via "OK Google, talk to $myAppName waarna jouw app tevoorschijn komt en de conversatie overneemt. 
+
+### Bouw je eigen app
+
+Voor het maken van een Assistant app heb je een aantal dingen nodig:
+1. Een [Actions on Google ](https://developers.google.com/actions/) project
+2. Een Webhook in een programmeertaal naar keuze (Node.js webhook gehost op Firebase Cloud Functions)
+3. Een platform dat de logica van de conversatie verzorgt ([Dialogflow](https://dialogflow.com/))
+
+De meest gebruikte platformen hiervoor zijn Firebase Cloud Functions in combinatie met de Firebase Realtime Database en Dialogflow voor de conversatie logica.
```

Comment:
Misschien even noemen dat Dialogflow voorheen API.ai werd genoemd

Reply:
Vind ik niet relevant, zou alleen maar verwarrend werken lijkt me.

Reply:
Lijkt mij ook verwarrend, ervan uitgaand dat de lezer Dialoglow/api.ai nog niet (goed) kent.

Reply:
Het vooral bij het zoeken... Als je er echt iets mee gaat doen (Dialogflow), kom je regelmatig uit op een resource waar API.ai in de naam voorkomt. Dat kan een handige got'cha zijn lijkt me.

Daarnaast denk ik dat we als doelgroep "onze collega's" kunnen hanteren. Dus stel dat je deze informatie (het bouwen van een asssitant app) aan een collega uitlegt, is het dan relevant om de term "API.ai" te laten vallen? Zo niet, dan idd maar weglaten..

Reply:
Als ik het aan een collega zou uitleggen zou ik geen api.ai meer noemen. Verder denk ik dat Google ondertussen slim genoeg is resultaten over api.ai te tonen wanneer op dialogflow gezocht wordt.